Lib/compression/zstd/__init__.py

```diff
@@ -0,0 +1,293 @@
+"""Python bindings to Zstandard (zstd) compression library, the API style is
+similar to Python's bz2/lzma/zlib modules.
+"""
+
+__all__ = (
+    # From this file
+    "compressionLevel_values",
+    "get_frame_info",
+    "CParameter",
+    "DParameter",
+    "Strategy",
+    "finalize_dict",
+    "train_dict",
+    "zstd_support_multithread",
+    "compress",
+    "decompress",
+    # From _zstd
+    "ZstdCompressor",
+    "ZstdDecompressor",
+    "ZstdDict",
+    "ZstdError",
+    "get_frame_size",
+    "zstd_version",
+    "zstd_version_info",
+    # From zstd.zstdfile
+    "open",
+    "ZstdFile",
+)
+
+import enum
+import functools
+import dataclasses
+
+from compression.zstd.zstdfile import ZstdFile, open
+from _zstd import *
+
+import _zstd
+
+
+_ZSTD_CStreamSizes = _zstd._ZSTD_CStreamSizes
+_ZSTD_DStreamSizes = _zstd._ZSTD_DStreamSizes
+_train_dict = _zstd._train_dict
+_finalize_dict = _zstd._finalize_dict
+
+
+@dataclasses.dataclass(frozen=True)
+class _CompressionLevelValues:
+    default: int
+    min: int
+    max: int
+
+compressionLevel_values = _CompressionLevelValues(*_zstd._compressionLevel_values)
```

Comment:
For Python code, we should adhere to PEP 8's guidance on naming where possible
```suggestion
compression_level_values = _CompressionLevelValues(*_zstd._compressionLevel_values)
```

Reply:
This one is a bit tricky, and I probably should have called it out. The `compressionLevel` name derives from the `ZSTD_c_compressionLevel` value in C. Since the min and max are already available through `CParameter.compressionLevel.bounds()`, perhaps it would be better to just set a value like `compressionLevel_default` or something and only include the result of `ZSTD_defaultCLevel()`? Or even set that on `CParameter.compressionLevel`?

Reply:
> The `compressionLevel` name derives from the `ZSTD_c_compressionLevel` value in C.

As a general point, I still think we should avoid leaking details from the implementation language (C) into Python. Most/almost all of the users of `compression.zstd` won't have directly interacted with the C source of `compressionLevel`, and remembering why the naming is different will be an annoying quirk.

If you're strongly in favour of keeping the camelCase naming I won't block it, but I think we should think carefully here.

-----

> perhaps it would be better to just set a value like `compressionLevel_default`

`zstd.COMPRESS{,ION}_LEVEL_DEFAULT` could make sense, I guess the question is the value of exposing the number as a public attribute. A quick check shows that only `lzma` and `zlib` do so  (PRESET_DEFAULT & Z_DEFAULT_COMPRESSION), but `bz2` and `gzip` don't.

Reply:
> If you're strongly in favour of keeping the camelCase naming I won't block it, but I think we should think carefully here.

I'm not, I think it would be reasonable to rename to snake_case. @Rogdham interested in your thoughts though.

Reply:
Oh and
> zstd.COMPRESS{,ION}_LEVEL_DEFAULT could make sense, I guess the question is the value of exposing the number as a public attribute. A quick check shows that only lzma and zlib do so (PRESET_DEFAULT & Z_DEFAULT_COMPRESSION), but bz2 and gzip don't.

I think it is generally useful, people like to use these values when wrapping compression libraries in their own code I've seen across a few bugs. (it's nicer  to say `level=zstd.COMPRESSION_LEVEL_DEFAULT` than `level=None` in function definitions imo)

Reply:
> I think it would be reasonable to rename to snake_case. @Rogdham interested in your thoughts though

+1 on this (using snake case for constants is aligned with PEP8)

Reply:
Just pushed changes to reflect this :)